host/xtest/Makefile

```diff
@@ -40,7 +40,7 @@ endef
 WITH_OPENSSL ?= y
 ifeq ($(WITH_OPENSSL),y)
 CFLAGS += -DOPENSSL_FOUND=1 -DOPENSSL_API_COMPAT=10100
-LDFLAGS += -lcrypto
+LDFLAGS += -L$(OPENSSL_DIR)/lib -lcrypto
```

Comment:
Is this needed for the CMakefile too?

Reply:
I haven't tried compiling with cmake to really understand how that works and if that's needed or not.  Do you want me to compile with cmake as well ?

Reply:
No, that's not needed, the CI loop should cover the building.
We try to keep the makefiles in sync featurewise.

Reply:
This patch is adding one more path to be searched while linking libcrypto.so provided by openssl.  This patch is not needed in both traditional make and cmake if we use the build system recommended by optee.  It is needed only when we want to use a openssl library that is compiled outside of the optee recommended build system.  

Now, coming to your question.  To achieve the above mentioned functionality we need to make changes in CMakeLists.txt as well.  But the current CMakeLists.txt is written very specific to be working with optee build system.  For eg, using traditional Makefile, we use a build flag to define if we want to link openssl libraries or not.  But in CMakeLists.txt, it searches for the openssl package and it links the libraries only if it finds the package.  I will need to re-write the logic here if I want to add the same functionality to CMakeLists.txt.

I will not try to do the same for CMakeLists.txt as well as I don't have enough proficiency to write/refactor the cmake files in a proper way.